### ISSUE TITLE
RichText: Support for line height

### ISSUE THREAD

**Issue by verdverm** [fyne-io/fyne#4287]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

I'm trying to use RichText with Markdown, and the text is rather compressed, vertically, compare the images below

some other issues of note
- not all newlines are respected, see the **Documentation** heading
- bullets are not indented
- code blocks have the same color background
- links at end of line have underline all the way to end of line on hover

**Fyne**

![image](https://github.com/fyne-io/fyne/assets/1390600/34a0becd-d791-480c-9376-390ffaccfbf7)

**GitHub**

![image](https://github.com/fyne-io/fyne/assets/1390600/28ee29a3-fa92-491f-a135-d0dd7e4aa559)

### Is it possible to construct a solution with the existing API?

I don't think so, it seems internal to RichText

maybe a `yPad` here that would be the `LineHeight`?

https://github.com/fyne-io/fyne/blob/v2.4.0/widget/richtext.go#L726

### Describe the solution you'd like to see.

LineHeight in TextStyle, I think this would be helpful generally, to align with how different sized textual elements typically have different sized line heights

Maybe having padding or margin on a text element can solve this?

I wonder if we might use something like `goldmark` like Hugo, with a custom walker that emits RichTextSegments?

**Comment by andydotxyz** [comment]
Are you looking for the `LineSpacing` size from theme? If you are in the code you'll see how that determines y position in the `Layout` loop that calls the row positioning helpers.

**Comment by verdverm** [comment]
No, that would be uniform, whereas the spacing per TextSegment type would allow, in example...

- lines in a paragraph vs between paragraphs
- bigger headings have more space after them
- bullets before/after vs between

This is similar to similar html elements having different line spacing rules

**Comment by andydotxyz** [comment]
> lines in a paragraph vs between paragraphs

This should already be supported (and demonstrable) as we have different line height and paragraph gap settings.

**Comment by verdverm** [comment]
I think the broader point is that we need more fine grained control, rather than a global setting that applies to everything.

If this is going to be an alternative to things like Electron and Tuari, then we will need more CSS like features on a per element basis.

**Comment by andydotxyz** [comment]
> I think the broader point is that we need more fine grained control, rather than a global setting that applies to everything.

There is something between global and controlling individual elements - and that is attaching behaviour to types of rich text elements...

To move the conversation forward I think it would be important to get a list of the "semantic spacings" so we can look at which are missing, and which are not set up correctly so we can address.

> Can you provide an example or link to, for example, how I can control the indentation of the bullet points in a list?

Unfortunately not, the indent is set by the text only - no padding value. We could move to a padding set in the theme for this if it is important.

**Comment by andydotxyz** [comment]
per-widget customising of padding will break alignment very fast.
There is work to allow special instances of theme overrides but exposing this type of customisation in the standard widgets is not going to be supported.

Keeping open for the headline line height support. Probably baked into the semantic types that RichText understands rather than a per-instance customisation, in-keeping with our semantic/behaviour API.
